zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/ReschedulingTask.java

```
@@ -70,13 +76,17 @@ private long onWorkPerformed(final int count) {
   }
 
   private long onError(final Throwable error) {
+    final String logMessage =
+        "Error occurred while performing a background task; operation will be retried";
     errorDelayMs = errorStrategy.applyAsLong(errorDelayMs);
 
-    // TODO: it's likely in some cases, we do want to log things as errors, but we need to
-    //  distinguish this from "normal" cases (e.g. missing data in ES, or ES temporarily
-    //  unavailable, etc.)
-    logger.warn(
-        "Error occurred while performing a background task; operation will be retried", error);
+    if (BACKGROUND_SUPPRESSED_EXCEPTIONS.contains(error.getCause().getClass())
+        && !logger.isDebugEnabled()) {
```

❓ Why we check for debugEnabled?

---

I think we should not bind this to debug, I guess it is in general still interesting to see when we can't connect/communicate (we shouldn't silently fail in my opinion). 

As it is a transient error, warning makes sense to me, it is a bigger issue if it happens for longer time, then some Operator can react and see the issues in their logs. 

It is not an error as there is nothing the Operator has to do _immediately_, as it might happen from time to time.

---

The context on which these `BACKGROUND_SUPPRESSED_EXCEPTIONS` can occur is pretty vast and vague IMO. Having the option to expand the stacktraces of these issues in case something else is acting out seems a good option to have. But I guess these could also happen throughout the exporter so they will be caught elsewhere.